static/src/javascripts/projects/common/modules/commercial/contributions-utilities.js

```diff
@@ -240,6 +240,74 @@ const articleViewCountIsOk = (
     return true;
 };
 
+const emitBeginEvent = (trackingCampaignId: string) => {
+    mediator.emit('register:begin', trackingCampaignId);
+};
+
+const emitInsertEvent = (
+    parentTest: EpicABTest,
+    products: $ReadOnlyArray<OphanProduct>,
+    campaignCode: ?string
+) => {
+    mediator.emit(parentTest.insertEvent, {
+        componentType: parentTest.componentType,
+        products,
+        campaignCode: campaignCode || '',
+    });
+};
+
+const setupOnView = (
+    element: HTMLElement,
+    parentTest: EpicABTest,
+    campaignCode: ?string,
+    trackingCampaignId: string,
+    products: $ReadOnlyArray<OphanProduct>,
+    showTicker: boolean = false
+) => {
+    // top offset of 18 ensures view only counts when half of element is on screen
+    const inView = elementInView(element, window, {
+        top: 18,
```

Comment:
Out of interest, where does the `18` here come from? I wonder if it's worth giving that value a name to clarify.

Reply:
Oh, I just noticed this was moved from further down (so not really introduced here).